src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ADD_PROPERTIES = 'addProperties';
+  const REMOVE_PROPERTIES = 'removeProperties';
+  const CLEAR = 'clear';
+
+  const statesArr = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    const { type: actionType } = action;
```

Comment:
So, what about the other properties? 
```suggestion
    const { type: actionType, extraData, keysToRemove } = action;
```